ext/mysql2/extconf.rb

```diff
@@ -69,7 +76,8 @@ def asplode lib
   libs = ['m', 'z', 'socket', 'nsl', 'mygcc']
   while not find_library('mysqlclient', 'mysql_query', lib, "#{lib}/mysql") do
     exit 1 if libs.empty?
-    have_library(libs.shift)
+    lib = libs.shift
+    asplode(lib) unless have_library(lib)
```

Comment:
Since this is in a loop, if `have_library` fails on the first iteration the rest won't be attempted since we're raising inside of `asplode` right?